### ISSUE TITLE
[ Expo Notifications ] [ ios ] Permissions undetermined after granting explicitly

### ISSUE THREAD

**Issue by actuallymentor** [expo/expo#8245]
### Environment

Target: ios. Not an issue on web or android.

### Steps to Reproduce

Use `Permissions.askAsync( Permissions.NOTIFICATIONS )`.

[See this snack](https://snack.expo.io/@mentor/vigorous-edamame).

### Expected Behavior

`{ status: 'granted' }`

### Actual Behavior

`{ status: 'undetermined' }`

### Reproducible Demo

[vigorous edamame](https://snack.expo.io/@mentor/vigorous-edamame).

**Comment by byCedric** [comment]
Hi @actuallymentor! Thanks for the issue report. [Could it be related to this](https://forums.expo.io/t/ios-push-notification-permission-status-undetermined/19395)? You have to set up some additional things on iOS for notifications to work, I'm not sure if this works from snack. If you have a project, you could try the solutions in the forum link. Let me know if that helps 😄

**Comment by actuallymentor** [comment]
Hey!

This is regarding a local SDK 37 project not yet published. There is no
reason it should be undetermined as per the documentation. If it is
expected behavior that ought to be documented.

I should add I've not had this problem before.

AFAIK something not working in snack is a red flag unless it's explicitly
not simulator compatible (which notification permissions ought to be).

**Comment by leekuo** [comment]
Seeing the exact same problem for local notification permissions on iOS simulator.  The status is never "granted" but always "undetermined".  The individual notification permissions like "allowsBadge", "allowsSound", "allowsAlert" seem to be returned.  The same problem with askAsync also affects the getAsync function results.

**Comment by actuallymentor** [comment]
@byCedric you tagged this `status: needs more info`, please indicate what info I can supply to help.

**Comment by actuallymentor** [comment]
@stalebot Noooooooooooo <cries in skywalker />

I'm happy to give more info but there is no clarity on what info is needed.

**Comment by sjchmiela** [comment]
Hey @actuallymentor! Sorry for taking so much time to respond to this issue.

What SDK version/environment are you testing the code on? If I remember correctly on some older SDKs `status` was being set to `"undetermined"` if the app wasn't able to register for remote notifications (which is the case in the simulator).

**Comment by actuallymentor** [comment]
No worries, with 715 open issues I get things can take time :)

It was with the most recent SDk at the time. Is that enough info or do you need a reproducible demo?

**Comment by sjchmiela** [comment]
What I meant is that I believe we've made some substantial changes to notifications code since then so this issue may have been resolved. Do you still experience it on latest SDK?

**Comment by sjchmiela** [comment]
Ok, I think I got to the bottom of the issue — basically it all comes down to the difference between `Permissions.USER_FACING_NOTIFICATIONS` and `Permissions.NOTIFICATIONS`.

The former one corresponds only to permissions related to showing the notifications while the latter one is like a compound permission encapsulating both `USER_FACING_NOTIFICATIONS` **and** whether device is registered for push notifications. Actually if you look into the [documentation for `Permissions.NOTIFICATIONS`](https://docs.expo.io/versions/latest/sdk/permissions/#permissionsnotifications) you'd see:

> **Note (iOS):** This does not disambiguate `undetermined` from `denied` and so will only ever return `granted` or `undetermined`. This is due to the way the underlying native API is implemented.

If you look into the implementation of the requester responsible for `.NOTIFICATIONS` you'd see:

https://github.com/expo/expo/blob/483db655aee826a564f7fbf5626ec8a1afcc5d08/packages/expo-notifications/ios/EXNotifications/Permissions/EXLegacyRemoteNotificationPermissionRequester.m#L41-L43

and since push notifications aren't supported on iOS simulator and it can't ever register for push notifications… you know what response will be always given.

To sum up:

- a slightly better way to fetch information about notifications permissions is to use `Permissions.USER_FACING_NOTIFICATIONS` which corresponds only to the notification permissions
- a much better way to fetch information about notifications permissions is to use `Notifications.getPermissionsAsync()` (the one described above could become deprecated in the future, we've moved away from having all the permissions in one package in favor of having them in corresponding packages)
- on iOS simulator you'll always get `undetermined` for `Permissions.NOTIFICATIONS` unless iOS simulators start supporting push notifications
- I'll add a note to the permissions docs that will make it even clearer that `undetermined` is what you should expect from iOS simulators.
